control/server.py

```
@@ -773,18 +773,56 @@ def _stop_spdk(self):
                 self.logger.exception(f"An error occurred while removing RPC "
                                       f"socket {self.spdk_rpc_socket_path}")
 
+    def _terminate_discovery(self, pid):
+        def is_running(pid):
+            try:
+                os.kill(pid, 0)
+                return True
+            except ProcessLookupError:
+                return False
+            except PermissionError:
+                self.logger.exception(f"Permission denied when checking status of discovery {pid}")
+                return True
+
+        def wait_for_exit(pid):
```

maybe its more of a personal taste, so dont do as you believe is better, but I would implement it with no return value, but rather raise a timeout exception when the process is running after the timeout. again, just personal taste as I believe this is the more commong behavior in such functions

---

sounds reasonable. for now waiting constants are defined here - https://github.com/baum/ceph-nvmeof/blob/4109b63818419f096f76e50aa9f8ee041a203e10/control/server.py#L788-L789
we might want to extract those into conf options in the future. diffs are welcome 🤗